### PR TITLE
Add newly supported Config resource types

### PR DESCRIPTION
**Pull request by obobrova**
*No issue number*

*Updated the list of resource types available to AWS Config's configuration recorder. The list was pulled from an existing Config recorder with all resources enabled. Also supported by the documentation: https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html#supported-resources*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json [hunk 1]
```diff
@@ -19,16 +24,23 @@
...
         "AWS::EC2::EIP",
+        "AWS::EC2::EgressOnlyInternetGateway",
+        "AWS::EC2::FlowLog",
         "AWS::EC2::Host",
         "AWS::EC2::Instance",
         "AWS::EC2::InternetGateway",
+        "AWS::EC2::NatGateway",
         "AWS::EC2::NetworkAcl",
         "AWS::EC2::NetworkInterface",
+        "AWS::EC2::RegisteredHAInstance",
```
Comment (kddejong) [inline src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json]: I can't seem to find this one...

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json [hunk 2]
```diff
@@ -71,6 +88,11 @@
         "AWS::WAFRegional::Rule",
         "AWS::WAFRegional::RuleGroup",
         "AWS::WAFRegional::WebACL",
+        "AWS::WAFv2::IPSet",
```
Comment (kddejong) [inline src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json]: Also not seeing waf2.  Is this documentation not updated?

File: src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json (hunk 1 above)
Comment (obobrova) [reply src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_config.json]: Right. A number of them are not there, namely:

* AWS::Config::ResourceCompliance
* AWS::EC2::RegisteredHAInstance
* AWS::WAFv2::IPSet
* AWS::WAFv2::ManagedRuleSet
* AWS::WAFv2::RegexPatternSet
* AWS::WAFv2::RuleGroup
* AWS::WAFv2::WebACL

However, they:
1. exist in the web console
2. are handled correctly by CloudFormation deployments

This is my first ever contribution to OSS. Please advise on what would be an acceptable proof of correctness here.

**Comment by kddejong** [comment]
Double checked and they are in the console.  Thanks @obobrova

**Pull request merged**